schemes/musica/tuvx/musica_ccpp_tuvx_cloud_optics.F90

```diff
@@ -0,0 +1,128 @@
+    use musica_tuvx_grid,     only: grid_t
+    use musica_tuvx_radiator, only: radiator_t
+    use musica_util,          only: error_t
+
+    type(grid_t),                        intent(inout) :: height_grid
+    type(grid_t),                        intent(inout) :: wavelength_grid
+    character(len=*),                    intent(out)   :: errmsg
+    integer,                             intent(out)   :: errcode
+    type(radiator_t), pointer                          :: radiator
+
+    ! local variables
+    type(error_t) :: error
+
+    num_vertical_levels = height_grid%number_of_sections( error )
+    if ( has_error_occurred( error, errmsg, errcode ) ) return
+
+    num_wavelength_bins = wavelength_grid%number_of_sections( error )
+    if ( has_error_occurred( error, errmsg, errcode ) ) return
+
+    radiator => radiator_t( cloud_optics_label, height_grid, wavelength_grid, &
+                            error )
+    if ( has_error_occurred( error, errmsg, errcode ) ) return
+
+  end function create_cloud_optics_radiator
+
+  !> Sets TUV-x cloud optics values
+  subroutine set_cloud_optics_values( radiator, cloud_fraction, delta_pressure, &
+                                      cloud_liquid_water_content, &
+                                      reciprocal_of_gravitational_acceleration, &
+                                      errmsg, errcode )
+    use ccpp_kinds,           only: kind_phys
+    use musica_ccpp_util,     only: has_error_occurred
+    use musica_tuvx_radiator, only: radiator_t
+    use musica_util,          only: error_t
+
+    type(radiator_t), intent(inout) :: radiator
+    real(kind_phys),  intent(in)    :: delta_pressure(:) ! pressure delta about vertical level midpoints (Pa)
+    real(kind_phys),  intent(in)    :: cloud_fraction(:) ! (unitless)
+    real(kind_phys),  intent(in)    :: cloud_liquid_water_content(:) ! (kg/kg)
+    real(kind_phys),  intent(in)    :: reciprocal_of_gravitational_acceleration ! (s^2/m)
+    character(len=*), intent(out)   :: errmsg
+    integer,          intent(out)   :: errcode
+
+    ! local variables
+    type(error_t)   :: error
+    real(kind_phys) :: od(num_vertical_levels)
+    real(kind_phys) :: cloud_optical_depth(num_vertical_levels, num_wavelength_bins)
+    integer         :: n_host_midpoints
+    integer         :: i_level
+    
+    n_host_midpoints = size(cloud_fraction)
+    if ( size(delta_pressure) /= n_host_midpoints ) then
+      errmsg = "[MUSICA Error] Invalid size of cloud pressure delta for TUV-x."
+      errcode = 1
+      return
+    end if
+    if ( size(cloud_liquid_water_content) /= n_host_midpoints ) then
+      errmsg = "[MUSICA Error] Invalid size of cloud liquid water content for TUV-x."
+      errcode = 1
+      return
+    end if
+    if ( n_host_midpoints + 1 /= num_vertical_levels ) then
+      errmsg = "[MUSICA Error] Invalid size of cloud fraction for TUV-x."
+      errcode = 1
+      return
+    end if
```

Comment:
the last block is what was checking that the size of `cloud_fraction` is correct, but maybe the `n_host_midpoints` was making this unclear. I've removed the `n_host_midpoints` and replaced it with `size(cloud_fraction)`. Let me know if this looks better.

Reply:
I see. Do you think it makes sense to check the cloud fraction size first, since the other two arrays are attempting to compare sizes under the assumption that the cloud fraction size is correct? Also, since the `size` function called five times, I think it might be better to define the variables as in the previous version.

Reply:
Good idea. updated!